django/contrib/admin/checks.py

```diff
@@ -901,21 +901,24 @@ def _check_list_display_item(self, obj, item, label):
             try:
                 field = getattr(obj.model, item)
             except AttributeError:
-                return [
-                    checks.Error(
-                        "The value of '%s' refers to '%s', which is not a "
-                        "callable, an attribute of '%s', or an attribute or "
-                        "method on '%s'."
-                        % (
-                            label,
-                            item,
-                            obj.__class__.__name__,
-                            obj.model._meta.label,
-                        ),
-                        obj=obj.__class__,
-                        id="admin.E108",
-                    )
-                ]
+                try:
+                    field = get_fields_from_path(obj.model, item)[-1]
+                except (FieldDoesNotExist, NotRelationField):
+                    return [
+                        checks.Error(
+                            "The value of '%s' refers to '%s', which is not a "
+                            "callable, an attribute, field of '%s', "
+                            "or an attribute or method on '%s'."
```

Comment:
I think this error message needs re-working. As proposed, and using the example from the tests, the error reads:

```
"The value of 'list_display[2]' refers to 'parent__foo', which is not a callable, an attribute, field of 'GrandChildAdmin', or an attribute or method on 'testapp.GrandChild'."
```

The proposed adding "field of" when used with the Admin class name is confusing, since this branch is adding support for model field lookups. So perhaps a better message would be:

```
"The value of 'list_display[2]' refers to 'parent__foo', which is not a callable, an attribute of 'GrandChildAdmin', or an attribute, method or Field on 'testapp.GrandChild'."
```
What do you think?

Reply:
How about:

> list_display[2], 'parent__foo', is not a callable, attribute of 'GrandChildAdmin', or attribute or lookup of 'testapp.GrandChild'.

"attribute" covers methods and fields too, so I think we're safe saying "attribute or lookup", and letting the docs cover the full list of supported values.